### PR TITLE
Expose field extension degree

### PR DESCRIPTION
**Pull request by AntoineRondelet**

### CODE REVIEW

**Comment by AntoineRondelet** [comment]
Related to https://github.com/clearmatics/zeth/pull/220

File: libff/algebra/fields/fp.hpp [hunk 1]
```diff
@@ -50,6 +50,8 @@ class Fp_model {
     static long long sqr_cnt;
     static long long inv_cnt;
 #endif
+    static const size_t tower_extension_degree = 1;
```
Comment (dtebbs) [inline libff/algebra/fields/fp.hpp:53]: Could make sense to use `constexpr` here, just for the avoidance of any doubt.

**Review by dtebbs — COMMENTED** [review]
I think it makes sense to re-review this once the use-cases in zeth have been finalized.  In particular we may want to remove the `tower_extension_degree` etc from Fp_model since there is no array to recurse through - but I'd need to actually check that the recursion can be done completely statically in the way I'm hoping.  clearmatics/zeth#220.

**Comment by AntoineRondelet** [comment]
> I think it makes sense to re-review this once the use-cases in zeth have been finalized. In particular we may want to remove the `tower_extension_degree` etc from Fp_model since there is no array to recurse through

You mean switching the coeffs to be of type `std::array<>` and accessing the number of coefficients as `el.coeffs.size()` rather than through the static attribute `tower_extension_degree`? Or do you mean removing the `extension_degree()` method?

**Comment by dtebbs** [comment]
> You mean switching the coeffs to be of type std::array<> and accessing the number of coefficients as el.coeffs.size() rather than through the static attribute tower_extension_degree? Or do you mean removing the extension_degree() method?

Ahk sorry - I wasn't clear.  I think the C-style array is perfectly fine.  I'm just thinking about the base field case, where the current version has `tower_extension_degree` but no array to iterate through.  It's kind of a subtle thing, but it seems to me that when recursing through the (extension) field types, we probably need to terminate based on type == `Fp_model` rather than `tower_extension_degree`==1, since we likely then need to call some method or use a member specific to `Fp_model`.

However, it would be good to have all the generic code in place in the zeth PR (so we can write generic code that works for `bw6_761_G2` as well), and then come back to this, since I'm pretty sure that will raise some issues I' havent' thought of.

**Comment by AntoineRondelet** [comment]
Superseded by #26

**Pull request closed without merging**